src/checkbox/checkbox.component.spec.ts

```diff
@@ -81,4 +95,13 @@ describe("Checkbox", async() => {
 		fixture.detectChanges();
 		expect(wrapper.onIndeterminateChange).toHaveBeenCalled();
 	});
+
+	describe("PAL exported tests", () => {
+		// Get checkbox from the fixture
+		element = setupFixture.debugElement.query(By.css("ibm-checkbox"));
+		setupFixture.detectChanges();
+		const instance = new CheckboxExportedTest({});
+		// tslint:disable-next-line
+		new BDDTestParser((instance as any).tests, setupFixture.debugElement.nativeElement);
```

Comment:
It seems to me, that as long as the tests are reasonably self contained we shouldn't run into serious perf issues ... I'm having trouble finding any nodejs memory benchmarking tools, so I'm not sure we can actually investigate this as much as we would like 😞 In general I think we should make sure to keep the tests as flat as possible (minimize nesting) and take advantage of the parallel nature of most test tools to keep our perf in check. If we have functions that can be pulled out of the classes and just used when needed that can also help things.